### DISCUSSION TITLE
Berry-ThingSpeak - how to use the cloud service from Mathlab

### DISCUSSION THREAD

**Discussion by wjohn007** [arendst/Tasmota#20583] [Show and tell]
If you want to store, aggregate and visualize data without a smart home server at all,
 Mathlab's ThingSpeak service is very helpful. 

 The Berry driver 'ThingSpeak' makes it easy to use this cloud-service.

Once your values are stored into a channel, you can see them instantly using the charts

![grafik](https://github.com/arendst/Tasmota/assets/115527085/9940691c-fc8c-4035-bba5-1221ab169b39)

It is also possible to aggregate data using Mathlab Visualization. <br/>
![grafik](https://github.com/arendst/Tasmota/assets/115527085/95bcc174-6812-4576-9d43-103d46f54642)

You can  integrate the charts into Tasmota via 'iframe'
![grafik](https://github.com/arendst/Tasmota/assets/115527085/38f59651-b682-446e-8f41-1cc11ef27e41)

Find detailed information at [Berry-ThingSpeak](https://github.com/wjohn007/Berry-DallasTemperature#berry-thingspeak).

John

**Comment by sfromis** [comment]
Nice :smile: 

Took a peek, and noticed a detail at the end of `ThingSpeak.be`; your destructor can't be called as the act of having the class instance registered as a driver means that Tasmota is keeping a reference, precluding garbage collection. Of course, this is normally not important.

**Comment by wjohn007** [comment]
Thank you for your feedback and the note about the destructor.

I use the destructor for 'cleanup ' after unit-tests.

```be
# --- cleanup

dt.deinit()
dt=nil
```
